clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -292,17 +292,43 @@ public int writeTo(TransferableChannel destChannel, int offset, int length) thro
     /**
      * Search forward for the file position of the message batch whose last offset that is greater
      * than or equal to the target offset. If no such batch is found, return null.
+     * <p>
+     * The following logic is intentionally designed to minimize memory usage
+     * by avoiding unnecessary calls to {@link FileChannelRecordBatch#lastOffset()} for every batch.
+     * Instead, we use {@link FileChannelRecordBatch#baseOffset()} comparisons when possible, and only 
+     * check {@link FileChannelRecordBatch#lastOffset()} when absolutely necessary.
```

Comment:
The javadoc should explain the behavior of the method. This type of comment should be next to the actual code - it doesn't affect callers.

Reply:
@m1a2st could you please file a minor to address it?

Reply:
https://github.com/apache/kafka/pull/19257